src/midgard/util.cc

```diff
@@ -97,35 +97,28 @@ void trim_shape(float start,
                 std::vector<PointLL>& shape) {
```

Comment:
this function is the crux of the fix. the diff is very hard to read but the previous code relied on finding the appropriate distance along the shape and only then setting the coordinate (ie trimming). the problem with that approach is if you are off by some small floating point amount you can end up never finding the spot to place the start vertex. instead we refactor to stop when we hit the tolerance or if we hit the end and then assign the vertex regardless.

previously what would happen is you'd get the whole shape and now we get that tiny little segment at the end.